app/service/auth_svc.py

```diff
@@ -144,47 +177,106 @@ async def is_request_authenticated(self, request):
 
     """ PRIVATE """
 
-    @staticmethod
-    async def _check_credentials(user_map, username, password):
-        user = user_map.get(username)
-        if not user:
-            return False
-        return user.password == password
-
-    async def _ldap_login(self, username, password):
-        server = ldap3.Server(self.ldap_config.get('server'))
-        dn = self.ldap_config.get('dn')
-        user_attr = self.ldap_config.get('user_attr') or 'uid'
-        user_string = '%s=%s,%s' % (user_attr, username, dn)
-
-        try:
-            with ldap3.Connection(server, user=user_string, password=password) as conn:
-                if conn.bind():
-                    if username not in self.user_map:
-                        group = await self._ldap_get_group(conn, dn, username, user_attr)
-                        await self.create_user(username, None, group)
-                    return True
-        except LDAPException:
-            self.log.error('Unable to connect to LDAP server')
+    def _set_primary_login_handler(self):
+        """
+        Returns the login handler for the auth service as specified in the config file.
+        This login handler will take priority for login methods during login_user and redirects during
+        check_permissions. If no login handler was specified in the config file, the default handler will be returned.
+        """
+        login_handler_module_path = self.get_config('auth.login.handler.module')
```

Comment:
Perhaps not needed in this PR, but I think it'd be good to move away from these string literals for configuration lookups. For example, having them defined as constants in a module somewhere, which would avoid mistakes via typos. Lotsa ways we could go, but the literals catch my eye whenever I see them.

Reply:
I'll put it as a constant in the current module for now